### DISCUSSION TITLE
BT BMS Battery LIFEPO

### DISCUSSION THREAD

**Discussion by shoties** [theengs/decoder#164] [New devices/things integration]
Hi
Could you be so kind to look at this BMS logs. 
Energy storage goes verry popular this day, could be cool to support such products in Home Assistant.
log is here: https://pastebin.pl/view/e7063299
![spagh](https://user-images.githubusercontent.com/72252829/181474661-f57c822c-80aa-42d2-aa2b-e8bf836e20ca.jpg)
![2022-07-28_11-35-57](https://user-images.githubusercontent.com/72252829/181474715-f684698a-cfad-4fcd-89a3-7019c0eb600e.jpg)

Thank You

**Comment by DigiH** [comment]
Hi @shoties,

this looks very interesting, but as Decoder is only able to process and decode freely broadcast data, and cannot connect to any device to get any further data, the only relevant information is in this screenshot

![Untitled 2](https://user-images.githubusercontent.com/17110652/181500524-2a1e378a-c9c4-4059-8d32-d8d34cb58dc8.png)

If you can get more of this information, with screenshots/written data to what information the app shows at **exactly** the same time, we can try and decode what is being broadcast, but I'm sure it won't be the full information.

For getting all the information, you might need to look into some other solution, similar to the user in

https://github.com/theengs/decoder/discussions/143

**Reply by shoties** [reply]
Was busy recently. Here is logs and ss. Thank you for Youre work
[Log 2.txt](https://github.com/theengs/decoder/files/9228774/Log.2.txt)
[Log bms.txt](https://github.com/theengs/decoder/files/9228775/Log.bms.txt)
![Screenshot_2022-07-31-13-40-34-724_com ble volyas](https://user-images.githubusercontent.com/72252829/182025071-0b90b92e-9e0a-4517-b1bb-c0aac5dc2b9a.jpg)
![Screenshot_2022-07-31-13-40-45-296_com ble volyas](https://user-images.githubusercontent.com/72252829/182025073-5266638a-6a5b-460d-a83b-c021dffd2ed5.jpg)
![Screenshot_2022-07-31-13-40-57-735_com ble volyas](https://user-images.githubusercontent.com/72252829/182025074-e4503877-3cba-432a-a75c-3a0224cecf89.jpg)
![Screenshot_2022-07-31-13-41-12-102_com ble volyas](https://user-images.githubusercontent.com/72252829/182025076-1b115cb4-b9fa-4e37-9276-8878c1b5191c.jpg)
![Screenshot_2022-07-31-13-41-20-118_com ble volyas](https://user-images.githubusercontent.com/72252829/182025077-9b4e59e3-33c9-45da-8661-05cee42e4a51.jpg)
![Screenshot_2022-07-31-13-41-27-094_com ble volyas](https://user-images.githubusercontent.com/72252829/182025080-54b2b05f-a215-40ca-8de7-6403358e415e.jpg)
![Screenshot_2022-07-31-13-41-34-155_com ble volyas](https://user-images.githubusercontent.com/72252829/182025081-af1e2891-4df4-422d-a1a3-cea2f1f50cc0.jpg)
![Screenshot_2022-07-31-13-43-58-909_no nordicsemi android mcp](https://user-images.githubusercontent.com/72252829/182025082-6ac4d231-9190-464f-89d5-5f80b6116b66.jpg)
![Screenshot_2022-07-31-13-44-05-272_no nordicsemi android mcp](https://user-images.githubusercontent.com/72252829/182025083-0f1e3f4e-62c1-4de2-a23b-62b2db6bc19d.jpg)

**Comment by DigiH** [comment]
Thank you, but I'm sorry, Decoder won't be able to get any information from the freely broadcast message, indicated in red below

![181500524-2a1e378a-c9c4-4059-8d32-d8d34cb58dc8](https://user-images.githubusercontent.com/17110652/182032538-08e6d9b8-5d56-40da-aeb3-1f9202d94f4e.png)

as it is just the MAC address (called Serial Number in the screenshot) in reverse.

![182025081-af1e2891-4df4-422d-a1a3-cea2f1f50cc0](https://user-images.githubusercontent.com/17110652/182032672-f41f19fe-e16e-4825-b500-4d9f5232ae2f.jpg)

To get any of the other data you'll have to find some other solution which can **connect** to the device to read the appropriate services/characteristics (as defined in your logs) to get the details you want. 

All the best
